pkg/cloud/azure/azure.go

```diff
@@ -85,3 +110,71 @@ func NewProviderAssets(config config.OperatorConfig) (common.CloudProviderAssets
 	}
 	return assets, nil
 }
+
+func IsAzure(infra *configv1.Infrastructure) bool {
+	if infra.Status.PlatformStatus != nil {
+		if infra.Status.PlatformStatus.Type == configv1.AzurePlatformType &&
+			(infra.Status.PlatformStatus.Azure.CloudName != configv1.AzureStackCloud) {
```

Comment:
Why do we exclude ASH from this? Might want a comment to explain that ASH is handled separately in its own transformer